packages/core/src/main/util/TaskScheduler/TaskScheduler.ts

```diff
@@ -48,13 +51,26 @@ const addTask = ({task, firingDate, key}: ScheduleTaskParams) => {
     cancelTask(key);
   }
 
-  const timeout = setTimeout(
-    async () => {
-      await task();
-      delete activeTimeouts[key];
-    },
-    delay > 0 ? delay : 0,
-  );
+  let timeout: ReturnType<typeof setTimeout>;
+
+  if (delay <= MAX_TIMEOUT_DELAY_VALUE) {
```

Comment:
When we try to pass delay higher than [max delay value (~28.4 days)](https://developer.mozilla.org/en-US/docs/Web/API/setTimeout#maximum_delay_value) the callback passed to `setTimeout` will execute immediately. To avoid it (but still support long delay times) in cases when delay time is too long for `setTimeout` we use `setInterval` instead and check every minute whether a task should be executed.

Reply:
How about we always use `setInterval` and try to avoid 2 different handling (that could create unexpected bugs when we reach a certain delay). 

I'd suggest you write a unit test that goes with, that would be a nice exercise

Reply:
Setting just one interval sounds good to me, but then I can't use TaskScheduler Adrian has created for committing pending proposals, because it has to use `setTimeout` for accuracy - what's also fine for me 👍 .

Reply:
Oh right, that makes sense. Maybe we could have 2 classes then:
- HighPrecisionTaskScheduler => more precise but with the performance issue that it create 1 timeout per task
- TaskScheduler => less precise but one has one single interval running

WDYT?

Reply:
Yeah, makes sense to me, both would be very useful, we'll decide which one to choose based how precisely we want the task to be executed 👍

Reply:
After digging into the code: Creating two task schedulers makes sense, but since we want to create single `setInterval` we won't be able to use it in our case. TaskScheduler is responsible for running a callback at specified `firingDate`. The thing is in this case we don't have single firingDate, we have many of them (one for each group), so there's no way to specify date at witch it should fire. The simplest solution would be just running single interval on client's init. What do you think?